File: HtOMSBrief/Shared/DashboardWidgetSnapshot.swift
import Foundation

/// 위젯에 전달하는 표시 전용 데이터. 인증정보나 사용자 식별자는 포함하지 않는다.
struct DashboardWidgetSnapshot: Codable, Equatable, Sendable {
    static let currentSchemaVersion = 1

    let schemaVersion: Int
    let todayAmount: Int
    let yesterdayAmount: Int
    let monthTotal: Int
    let previousMonthTotal: Int
    let dailyAverage: Int
    let targetAmount: Int
    let refreshedAt: Date
    let serverTime: Date
    let isSample: Bool

    static let sample: DashboardWidgetSnapshot = {
        var calendar = Calendar(identifier: .gregorian)
        calendar.timeZone = TimeZone(identifier: "Asia/Seoul") ?? .current
        let refreshedAt = calendar.date(from: DateComponents(
            year: 2026, month: 8, day: 20, hour: 16, minute: 48
        )) ?? Date(timeIntervalSince1970: 0)
        let serverTime = calendar.date(from: DateComponents(
            year: 2026, month: 8, day: 20, hour: 16, minute: 40
        )) ?? Date(timeIntervalSince1970: 0)

        return DashboardWidgetSnapshot(
            schemaVersion: currentSchemaVersion,
            todayAmount: 10_667_000,
            yesterdayAmount: 18_160_000,
            monthTotal: 203_527_398,
            previousMonthTotal: 320_595_397,
            dailyAverage: 10_791_012,
            targetAmount: 307_482_479,
            refreshedAt: refreshedAt,
            serverTime: serverTime,
            isSample: true
        )
    }()
}

File: app/src/main/java/com/htoms/brief/model/DashboardWidgetSnapshot.kt
package com.htoms.brief.model

import kotlinx.serialization.Serializable
import java.util.Calendar
import java.util.TimeZone

/**
 * 위젯에 전달하는 표시 전용 데이터. 인증정보나 사용자 식별자는 포함하지 않는다.
 */
@Serializable
data class DashboardWidgetSnapshot(
    val schemaVersion: Int = CURRENT_SCHEMA_VERSION,
    val todayAmount: Long,
    val yesterdayAmount: Long,
    val monthTotal: Long,
    val previousMonthTotal: Long,
    val dailyAverage: Long,
    val targetAmount: Long,
    val refreshedAt: Long,
    val serverTime: Long,
    val isSample: Boolean
) {
    companion object {
        const val CURRENT_SCHEMA_VERSION = 1

        val sample: DashboardWidgetSnapshot by lazy {
            val calendar = Calendar.getInstance(TimeZone.getTimeZone("Asia/Seoul")).apply {
                set(2026, Calendar.AUGUST, 20, 16, 48, 0)
                set(Calendar.MILLISECOND, 0)
            }
            val refreshedAt = calendar.timeInMillis

            calendar.apply {
                set(2026, Calendar.AUGUST, 20, 16, 40, 0)
                set(Calendar.MILLISECOND, 0)
            }
            val serverTime = calendar.timeInMillis

            DashboardWidgetSnapshot(
                schemaVersion = CURRENT_SCHEMA_VERSION,
                todayAmount = 10_667_000L,
                yesterdayAmount = 18_160_000L,
                monthTotal = 203_527_398L,
                previousMonthTotal = 320_595_397L,
                dailyAverage = 10_791_012L,
                targetAmount = 307_482_479L,
                refreshedAt = refreshedAt,
                serverTime = serverTime,
                isSample = true
            )
        }
    }
}
